Command/PasswordResetHandler.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\Command;
+
+use SumoCoders\FrameworkMultiUserBundle\Exception\InvalidPasswordConfirmationException;
+use SumoCoders\FrameworkMultiUserBundle\User\UserInterface;
+use SumoCoders\FrameworkMultiUserBundle\User\UserRepositoryCollection;
+
+class PasswordResetHandler
+{
+    /**
+     * @var UserRepositoryCollection
+     */
+    private $userRepositoryCollection;
+
+    public function __construct(UserRepositoryCollection $userRepositoryCollection)
+    {
+        $this->userRepositoryCollection = $userRepositoryCollection;
+    }
+
+    /**
+     * @param PasswordReset $command
+     * 
+     * @throws InvalidPasswordConfirmationException
+     */
+    public function handle(PasswordReset $command)
+    {
+        if ($command->passwordConfirmationIsValid()){
+            $user = $command->getUser();
+            $updatedUser = clone $user;
+            $updatedUser->setPassword($command->getPassword());
+            $repository = $this->userRepositoryCollection->findRepositoryByClassName(get_class($user));
+            $repository->update($user, $updatedUser);
+            return;
+        }
+        
+        throw new InvalidPasswordConfirmationException('The password confirmation isn\'t valid');
```

Comment:
ik zou dit omgekeerd doen. De throw in de if en dan de rest er na. Dat zal wat vlotter lezen. (minder code in een indentatie)